pkg/controller/volume/attachdetach/reconciler/reconciler_test.go

```diff
@@ -834,7 +833,7 @@ func Test_Run_OneVolumeAttachAndDetachTimeoutNodesWithReadWriteOnce(t *testing.T
 	// Volume is added to asw. Because attach operation fails, volume should not be reported as attached to the node.
 	waitForVolumeAddedToNode(t, generatedVolumeName, nodeName1, asw)
 	verifyVolumeAttachedToNode(t, generatedVolumeName, nodeName1, cache.AttachStateUncertain, asw)
-	verifyVolumeReportedAsAttachedToNode(t, logger, generatedVolumeName, nodeName1, false, asw)
+	verifyVolumeReportedAsAttachedToNode(t, logger, generatedVolumeName, nodeName1, false, asw, volumeAttachedCheckTimeout)
```

Comment:
So now - most calls to `verifyVolumeNoStatusUpdateNeeded` are prone to flaking, because the state it is checking depends on when goroutine spawned by reconciler for detaching volume finishes and when reconciler loop itself turns. 

I propose we remove all of those calls except the last one - `verifyVolumeNoStatusUpdateNeeded(t, logger, generatedVolumeName, nodeName2, asw)` 

I also think we should remove the intermediate checks - https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/volume/attachdetach/reconciler/reconciler_test.go#L756-L765  . I know they are useful to have but if a test flakes even inside a debugger, it will cause bad dev experience.

Reply:
I agree if there is no obvious benefit of checking intermediate state, we can remove those checks. Wondering it start to flake much more than before, and what has changed? Any performance implications?

Reply:
> I also think we should remove the intermediate checks -  https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/volume/attachdetach/reconciler/reconciler_test.go#L756-L765 . I know they are useful to have but if a test flakes even inside a debugger, it will cause bad dev experience.

I commented checking intermediate state code here.

> Any performance implications?

It flakes for a quite long time in my memory. But yes, it flake more recently than before(I prefer to think this is release related as there are more CI jobs in code freeze week and the CI load is larger.).